packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -113,6 +139,12 @@ export class WaferMap extends FoundationElement {
 
     public override connectedCallback(): void {
         super.connectedCallback();
+        const canvasContext = this.canvas.getContext('2d', {
+            willReadFrequently: true
+        });
+        if (canvasContext != null) {
+            this.canvasContext = canvasContext;
+        }
```

Comment:
The null check doesn't really give us any benefit here. The default value of the property in undefined due to:
 ```ts
public canvasContext!: CanvasRenderingContext2D;
```
So if getContext fails to run the value of `this.canvasContext` is still `undefined`. The null check doesn't make the application more robust, future code will still have an error.

Unless we want to write more robust recovery code (I'm not sure under what conditions getContext will fail or how to recover), we can just assert that it does not fail:

```suggestion
        this.canvasContext = this.canvas.getContext('2d', {
            willReadFrequently: true
        })!;
```